crates/bevy_math/src/interpolation/interpolate.rs

```diff
@@ -0,0 +1,190 @@
+use super::{utils, Lerp};
+use crate::{Quat, Vec2, Vec3, Vec3A, Vec4};
+
+// http://archive.gamedev.net/archive/reference/articles/article1497.html (bit old)
+
+#[derive(Debug, Copy, Clone)]
+pub struct TangentIgnore;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Interpolation {
+    Step,
+    Linear,
+    Hermite,
+}
+
+pub trait Interpolate: Lerp + Clone {
+    /// Tangent used for the hermite interpolation
+    type Tangent: Copy;
+
+    const FLAT_TANGENT: Self::Tangent;
+
+    #[inline]
+    fn interpolate(
+        k0: &Self,
```

Comment:
The variable names in these traits are hard to follow without any doc strings.

Reply:
Also, I wouldn't be surprised if this triggers a `clippy` `too-many-arguments` warning. 
I would suggest grouping the parameters into structs.
```rust
fn foo(a: u32, b: u32, c: i32, d: i32);

// vs

struct AB {
    a: u32,
    b: u32,
}

struct CD {
    c: i32,
    d: i32,
}

fn foo(ab: AB, cd: CD);
```
(granted the names in this example are bad)

Reply:
that will only add more boiler plate, clippy is happy

Reply:
done